### PR TITLE
Verify uncommitted new i18n statements

### PR DESCRIPTION
**Pull request by ferhoyos**
### Describe the change

When a developer adds a new i18n label (e.g., t('label')), translation files must be re-generated with the new i18n statements. To avoid committing without the new i18n statements, this PR updates the pre-commit hook to re-generate the translation files and ensure that no new i18n statements remain uncommitted.

### Steps to test the PR

1. Add a new i18n label (e.g., change `Log In` to `t('Log In')` in [LoginPage.tsx](https://github.com/kiali/kiali/blob/86d5778217d5784aeb8bfbe4f6d712980378a91f/frontend/src/pages/Login/LoginPage.tsx#L191)).
2. Attempt to create a commit.
3. Verify that the commit is aborted because new i18n statements have not been committed and `translation.json` is modified with the new i18n statement.

### Automation testing

N/A

### CODE REVIEW

File: hack/hooks/pre-commit.sh [hunk 1]
```diff
@@ -39,8 +39,20 @@ if [ -n "$yaml_unformatted" ]; then
...
+i18n_files=$(git diff --name-only HEAD --diff-filter=M | grep -E 'translation.json')
+
 #### Git commit check ####
 if [ -n "$yaml_unformatted" ] || [ -n "$go_unformatted" ]; then
   echo "Some files have been formatted - the git commit is aborted."
   exit 1
 fi
+
+if [ -n "$i18n_files" ]; then
+  echo "New i18n statements have not been committed - the git commit is aborted."
```
Comment (josunect) [inline hack/hooks/pre-commit.sh:56]: Could we add instructions about how to fix it? Now I know, but maybe someone doesn't have an idea about how to fix it. We could add (Ex): "Run 'make build-ui' and include the generated translation files" (Or a link with the instructions)?

**Review by josunect — COMMENTED** [review]
Thanks, that's very useful and it works as expected. 
image

I've just added a small comment.

File: hack/hooks/pre-commit.sh (hunk 1 above)
Comment (ferhoyos) [reply hack/hooks/pre-commit.sh:56]: With the verification the new i18n statements are added to the translation files, but not committed.

Is this message clearer for you?:

```
echo "New i18n statements are generated but not committed - the git commit is aborted. Please include the updated translation files in the commit"
```

Comment (josunect) [reply hack/hooks/pre-commit.sh:56]: Ah, i didn't realized it already was generated! So it looks good.

**Review by josunect — APPROVED** [review]

**Review by josunect — APPROVED** [review]
lgtm!

**Review by jmazzitelli — APPROVED** [review]
I approve. It works, and the new pre-commit check is fast, so it shouldn't bother anyone because the commit doesn't slow up much.

**Pull request merged**